### PR TITLE
Update substrait requirement from 0.12.0 to 0.13.1

### PR DESCRIPTION
**Pull request by viirya**
## Which issue does this PR close?

Closes #7447.

### CODE REVIEW

File: datafusion/substrait/src/logical_plan/producer.rs [hunk 1]
```diff
@@ -1378,6 +1383,15 @@ fn to_substrait_bound(bound: &WindowFrameBound) -> Bound {
     }
 }
 
+fn to_substrait_bound_type(window_frame: &WindowFrame) -> Result<BoundsType> {
+    match window_frame.units {
+        WindowFrameUnits::Rows => Ok(BoundsType::Rows), // ROWS
+        WindowFrameUnits::Range => Ok(BoundsType::Range), // RANGE
+        // TODO: Support GROUPS
+        unit => not_impl_err!("Unsupported window frame unit: {unit:?}"),
```
Comment (viirya) [inline datafusion/substrait/src/logical_plan/producer.rs:1391]: substrait `WindowFrameUnits` doesn't support GROUPS frame type.

File: datafusion/physical-expr/src/datetime_expressions.rs [hunk 2]
```diff
@@ -462,6 +462,7 @@ fn date_bin_months_interval(stride_months: i64, source: i64, origin: i64) -> i64
     bin_time.timestamp_nanos()
 }
 
+#[allow(deprecated)]
```
Comment (viirya) [inline datafusion/physical-expr/src/datetime_expressions.rs:465]:
```
   --> datafusion/physical-expr/src/datetime_expressions.rs:469:22
    |
469 |     DateTime::<Utc>::from_utc(date, Utc)
    |                      ^^^^^^^^
    |
    = note: `-D deprecated` implied by `-D warnings`
```

I think this is not from substrait upgrade, but CI complains about it.

File: datafusion/optimizer/tests/optimizer_integration.rs [hunk 3]
```diff
@@ -332,6 +332,7 @@ fn test_same_name_but_not_ambiguous() {
     assert_eq!(expected, format!("{plan:?}"));
 }
 
+#[allow(deprecated)]
```
Comment (viirya) [inline datafusion/optimizer/tests/optimizer_integration.rs:335]: ditto

**Comment by viirya** [comment]
cc @alamb

**Review by yjshen — APPROVED** [review]

File: datafusion/physical-expr/src/datetime_expressions.rs (hunk 2 above)
Comment (yjshen) [reply datafusion/physical-expr/src/datetime_expressions.rs:465]: https://github.com/apache/arrow-datafusion/pull/7451 tries to fix this.

Comment (viirya) [reply datafusion/physical-expr/src/datetime_expressions.rs:465]: Ah, okay, I will rebase after the fix is merged.

**Comment by viirya** [comment]
Thank you @yjshen

**Pull request merged**